substrate/client/rpc/src/lib.rs

```diff
@@ -45,3 +45,147 @@ pub mod testing;
 
 /// Task executor that is being used by RPC subscriptions.
 pub type SubscriptionTaskExecutor = std::sync::Arc<dyn sp_core::traits::SpawnNamed>;
+
+/// JSON-RPC helpers.
+pub mod utils {
+	use crate::SubscriptionTaskExecutor;
+	use futures::{
+		future::{self, Either, Fuse, FusedFuture},
+		Future, FutureExt, Stream, StreamExt,
+	};
+	use jsonrpsee::{PendingSubscriptionSink, SubscriptionMessage, SubscriptionSink};
+	use sp_runtime::Serialize;
+	use std::collections::VecDeque;
+
+	/// A simple bounded VecDeque.
+	struct BoundedVecDeque<T> {
+		inner: VecDeque<T>,
+		max_cap: usize,
+	}
+
+	impl<T> BoundedVecDeque<T> {
+		/// Create a new bounded VecDeque.
+		fn new(max_cap: usize) -> Self {
+			Self { inner: VecDeque::with_capacity(max_cap), max_cap }
+		}
+
+		fn push_back(&mut self, item: T) -> Result<(), ()> {
+			if self.inner.len() >= self.max_cap {
+				Err(())
+			} else {
+				self.inner.push_back(item);
+				Ok(())
+			}
+		}
+
+		fn pop_front(&mut self) -> Option<T> {
+			self.inner.pop_front()
+		}
+	}
+
+	/// Feed items to the subscription from the underlying stream.
+	///
+	/// It's possible configure how many items from the stream
+	/// are allowed to be kept in memory until the subscription is dropped.
+	///
+	/// This is needed because the underlying streams in substrate are
+	/// unbounded.
+	pub async fn pipe_from_stream<S, T>(
```

Comment:
This was provided by jsonrpsee before and I added something similar as a starting point.

The new rpc-spec has specified how to deal with if the server can't keep with the number of messages.
For now each implementation can specify how many item is allowed by kept in memory until a subscription is dropped and is this PR I used 16 but if anyone wants another number for a specific subscription it's doable as well.

This is needed because for instance import_streams and storage_streams are unbounded.

//cc @koute maybe you have some ideas for this until we abandon the old rpc APIs? :)